# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.h
# Commit: 7f259ef8550f
# Full Hash: 7f259ef8550fcce1495ae80747dc5d3309bb9de4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-24 21:44:48
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - After unmuting, re-initialize the AudioSink asynchronously. r=alwu
#   
#   On Windows (in particular, but also not quick on other OS, depending on the
#   hardware being used), opening a system-level audio stream (using cubeb)
#   can be very long. It's a blocking call from the MDSM thread. The MDSM can't
# ==============================================================================

diff -r 16633fa96586 -r 7f259ef8550f dom/media/mediasink/AudioSinkWrapper.h
--- a/dom/media/mediasink/AudioSinkWrapper.h	Tue May 24 13:09:09 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.h	Tue May 24 13:09:09 2022 +0000
@@ -103,7 +103,17 @@
     MOZ_ASSERT(mOwnerThread->IsCurrentThreadIn());
   }
 
-  nsresult StartAudioSink(const media::TimeUnit& aStartTime);
+  // An AudioSink can be started synchronously from the MDSM thread, or
+  // asynchronously.
+  // In synchronous mode, the clock doesn't advance until the sink has been
+  // created, initialized and started. This is useful for the initial startup,
+  // and when seeking.
+  // In asynchronous mode, the clock will keep going forward (using the system
+  // clock) until the AudioSink is started, at which point the clock will use
+  // the AudioSink clock. This is used when unmuting a media element.
+  enum class AudioSinkStartPolicy { SYNC, ASYNC };
+  nsresult StartAudioSink(const media::TimeUnit& aStartTime,
+                          AudioSinkStartPolicy aPolicy);
 
   // Get the current media position using the system clock. This is used when
   // the audio is muted, or when the media has no audio track. Otherwise, the
